server/management/documents/appointmentsFilingStatuses.php

```diff
@@ -0,0 +1,144 @@
+	header ('Pragma: public');
+
+	$objWriter = $phpExcelWrapper->createExcelWriter();
+	$objWriter->save('php://output');
+
+	exit;
+}
+
+function executeAppointmentQuery($data) {
+	GLOBAL $DB_CONN, $ALL_SITES_ID;
+
+	$query = 'SELECT Client.firstName, Client.lastName, Appointment.appointmentId,
+		ServicedAppointment.completed, ServicedAppointment.notCompletedDescription, ServicedAppointment.servicedAppointmentId,
+		Site.title
+		FROM Appointment
+		LEFT JOIN ServicedAppointment ON Appointment.appointmentId = ServicedAppointment.appointmentId
+		JOIN Client ON Appointment.clientId = Client.clientId
+		JOIN AppointmentTime ON Appointment.appointmentTimeId = AppointmentTime.appointmentTimeId
+		JOIN Site ON AppointmentTime.siteId = Site.siteId
+		WHERE DATE(AppointmentTime.scheduledTime) = ?
+			AND Appointment.archived = FALSE';
+	if ($data['siteId'] != $ALL_SITES_ID) {
+		$query .= ' AND AppointmentTime.siteId = ?';
+	}
+	$query .= ' ORDER BY AppointmentTime.siteId ASC, AppointmentTime.scheduledTime ASC';
+	$stmt = $DB_CONN->prepare($query);
+
+	$filterParams = array($data['date']);
+	if ($data['siteId'] != $ALL_SITES_ID) {
+		$filterParams[] = $data['siteId'];
+	}
+	$stmt->execute($filterParams);
+	$appointments = $stmt->fetchAll(PDO::FETCH_ASSOC);
+
+
+	$filingStatusQuery = 'SELECT lookupName 
+		FROM AppointmentFilingStatus
+		JOIN FilingStatus ON AppointmentFilingStatus.filingStatusId = FilingStatus.filingStatusId
+		WHERE AppointmentFilingStatus.servicedAppointmentId = ?';
+	$stmt = $DB_CONN->prepare($filingStatusQuery);
+	foreach ($appointments as &$appointment) {
+		$appointment['stateEFile'] = false;
+		$appointment['federalEFile'] = false;
+		$appointment['statePaper'] = false;
+		$appointment['federalPaper'] = false;
+
+		if (!isset($appointment['servicedAppointmentId'])) continue;
+
+		$stmt->execute(array($appointment['servicedAppointmentId']));
+		$filingStatuses = $stmt->fetchAll(PDO::FETCH_ASSOC);
+		foreach ($filingStatuses as $filingStatus) {
+			if ($filingStatus['lookupName'] === 'state_efile') $appointment['stateEFile'] = true;
+			else if ($filingStatus['lookupName'] === 'federal_efile') $appointment['federalEFile'] = true;
+			else if ($filingStatus['lookupName'] === 'state_paper') $appointment['statePaper'] = true;
+			else if ($filingStatus['lookupName'] === 'federal_paper') $appointment['federalPaper'] = true;
+		}
+	}
+
+	return $appointments;
+}
+
+function createAppointmentsFilingStatusesExcelFile($appointments) {
+	GLOBAL $HEADER_COLUMN_NAMES;
+	$phpExcelWrapper = new PHPExcelWrapper();
+
+	if (empty($appointments)) {
+		$sheetIndex = $phpExcelWrapper->createSheet('None');
+		$phpExcelWrapper->setActiveSheetIndex($sheetIndex);
+		$phpExcelWrapper->insertHeaderRow($HEADER_COLUMN_NAMES);
+		$phpExcelWrapper->nextRow();
+	} else {
+		# Iterate through all the results and append them to the proper sheet
+		$sheetIndexForSiteId = array();
+		foreach ($appointments as $row) {
+			$siteId = $row['siteId'];
```

Comment:
This line doesn't exactly work since the query isn't pulling a siteId. As is, if you try to download the document for all sites, it will only create a sheet for the first site.